File: back/src/main/java/csh/back/domain/trip/member/dto/response/TripMemberResponse.java
package csh.back.domain.trip.member.dto.response;

import csh.back.domain.trip.member.entity.TripMember;

public record TripMemberResponse(
		Long memberId,
		String name,
		boolean admin
) {
	public static TripMemberResponse from(TripMember tripMember) {
		return new TripMemberResponse(
				tripMember.getMember().getId(),
				tripMember.getMember().getName(),
				tripMember.isAdmin()
		);
	}
}

File: back/src/main/kotlin/csh/back/domain/trip/member/dto/response/TripMemberResponse.kt
package csh.back.domain.trip.member.dto.response

import csh.back.domain.trip.member.entity.TripMember

data class TripMemberResponse(
    val memberId: Long,
    val name: String,
    val admin: Boolean,
) {
    companion object {
        @JvmStatic
        fun from(tripMember: TripMember): TripMemberResponse = TripMemberResponse(
            memberId = requireNotNull(tripMember.member.id),
            name = tripMember.member.name,
            admin = tripMember.isAdmin,
        )
    }
}
